### PR TITLE
feat: Allow clients to access messages from eslint

### LINKED ISSUE #704: ability to capture (non-fixable) eslint errors/warnings?
**Versions**:

- `prettier-eslint` version: 13.0.0
- `node` version: 16.13.12
- `npm` (or `yarn`) version: npm 8.1.2

**Problem description:**

Currently, `prettier-eslint` explicitly discards errors/warnings about eslint rule violations, as it is focused just on amending code to be compliant where it can. But I use `prettier-eslint-cli` to run `prettier-eslint` automatically on save from my editor. In that use case, it would be convenient for such errors/warnings to be written to stderr -- if that happened, they would pop up in my editor when I saved a file, which would save a step of running eslint at the end to check for any unfixable problems.

**Suggested solution:**

Add an option to `prettier-eslint` to pass back eslint error/warning messages about rule violations, in addition to the amended code, so that `prettier-eslint-cli` could then (perhaps optionally) write those to stderr. (Note I have yet to file an issue about this on the `prettier-eslint-cli` repo because at the moment there's no way the information could be written to stderr, since `prettier-eslint` is not providing it.)

Thanks for considering.

### PR DESCRIPTION
**Pull request by gwhitney**
Adds another export `analyze` from this package, which behaves
  exactly like `format` except that it returns a simple object
  with properties `output` giving the formatted text and `messages`
  giving the eslint messages produced in processing the input text.
  Clients can call `analyze` instead of `format` if they wish, for
  example, to capture any errors that eslint may have encountered.

  Resolves #704.

### CODE REVIEW

**Comment by gwhitney** [comment]
I expect that given that this PR adds a new export, it should have a changeset. So I clicked on the link about how to add a changeset, and I tried `npx changeset` in the top-level directory of my clone, but it reports `could not determine executable to run`. I tried `sudo npm install -g changeset`, but still neither `npx changeset` nor `changeset` worked. So I am currently unable to add a changeset, but would be happy to do so if further guidance could be provided.

**Comment by idahogurl** [comment]
@gwhitney To create a changeset, the command should be `npx -p @changesets/cli changeset`

**Comment by gwhitney** [comment]
Thanks! Now it reports `🦋  error There is no .changeset folder.` so sadly I am still stuck.

**Comment by idahogurl** [comment]
@gwhitney If you want you can add me to as a contributor to your forked repo and I'll make a PR to put the changeset in

**Comment by gwhitney** [comment]
I'm confused? I've already allowed edits by maintainers...

**Comment by idahogurl** [comment]
@gwhitney https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-teams-and-people-with-access-to-your-repository#changing-permissions-for-a-team-or-person

**Comment by gwhitney** [comment]
I am still confused, why do you need access to the entire fork as opposed to this branch on the fork, which you presumably already have? That was my question, sorry I wasn't clear...

**Comment by gwhitney** [comment]
Thanks so much for doing that for me! I really appreciate it.

File: .eslintrc.js [hunk 1]
```diff
@@ -12,6 +12,9 @@ const config = {
       }
     ],
     'import/no-import-module-exports': 'off'
+  },
+  settings: {
+    'import/ignore': ['node_modules', 'src'] // Using CommonJS in src
```
Comment (idahogurl) [inline .eslintrc.js:17]: Why is this necessary?

File: README.md [hunk 2]
```diff
@@ -146,7 +146,46 @@ This allows you to use `eslint` to look for bugs and/or bad practices, and use
 
 `prettier-eslint` will **only** propagate _parsing_ errors when either `prettier` or `eslint` fails. In addition to propagating the errors, it will also log a specific message indicating what it was doing at the time of the failure.
 
-**Note:** `prettier-eslint` will not show any message regarding broken rules in either `prettier` or `eslint`.
+**Note:** `format` will not show any message regarding broken rules in either `prettier` or `eslint`.
+
+## Capturing eslint messages
```
Comment (idahogurl) [inline README.md:151]:
```suggestion
## Capturing ESLint messages
```

**Review by idahogurl — CHANGES_REQUESTED** [review]

File: .eslintrc.js (hunk 1 above)
Comment (gwhitney) [reply .eslintrc.js:17]: Because without it, eslint reports
```
/home/glen/code/prettier-eslint/src/__tests__/index.js
  8:9   error  format not found in '../'   import/named
  8:17  error  analyze not found in '../'  import/named
```
even though `../index.js` is perfectly well exporting format and analyze as the unit tests show. The difficulty is that the src directory of prettier-eslint is in CommonJS module format, and the `import/named` rule really only functions with ES modules. An alternative might be to turn off that one particular eslint rule, but since (short of switching pretter-eslint to ES modules, which would also be fine but beyond the scope of this PR) all import checking is doomed to fail, it seemed better to just let eslint know to ignore issues about imports from src altogether.

File: README.md (hunk 2 above)
Comment (gwhitney) [reply README.md:151]: On the first suggestion, I didn't change that occurrence because neither 'format' nor 'analyze' _show_ any messages (as in write them anywhere), just 'analyze' will pass them back. So the sentence seemed to still be correct. But I will be happy to change it if you like, just let me know. And I will change the second one once I hear about that, thanks.

Comment (idahogurl) [reply README.md:151]: @gwhitney It's only about changing `eslint` to `ESLint` as that is the name proper name not just the package name.

Comment (gwhitney) [reply README.md:151]: No, I get that but what about the first change you suggested of  "prettier-eslint" to "format"? I thought about that when I first submitted the PR but since neither format or analyze "shows" anything, i.e., they don't write any messages to the console or anything, I left it the way it was because it still seemed accurate. But I will change it to 'format' if you like. Please just let me know. Thanks.

Comment (idahogurl) [reply README.md:151]: @gwhitney Oh I see. I didn't suggest that. Not sure why it's showing up as a change.

**Comment by gwhitney** [comment]
I see the stale marking; is there anything I am supposed to be doing to advance this? Thanks for letting me know.

**Comment by gwhitney** [comment]
Why was this closed? Was it deemed not desirable for the project? Or would some different approach to issue #704 be preferred? Please let me know what I can do to work toward a solution for that issue. Thanks!

**Review by idahogurl — APPROVED** [review]

**Comment by JounQin** [comment]
Well, the stale bot is stupid in this case...

**Comment by JounQin** [comment]
@gwhitney Do you have anytime to fix the conflicts?

**Comment by gwhitney** [comment]
> @gwhitney Do you have anytime to fix the conflicts?

Hope to later today, not 100% certain. Will keep you posted.

**Comment by gwhitney** [comment]
OK, did my best to rebase. The new test runs OK on my machine. Let me know what else I can do to help. Thanks!

**Comment by JounQin** [comment]
I don't see anything strange, so merging.

**Comment by JounQin** [comment]
Released as [v16.2.0](https://github.com/prettier/prettier-eslint/releases/tag/v16.2.0).

Thanks for your contribution! @gwhitney
